### ISSUE TITLE
📣 Elementor v3.1 Beta 4 Release 📣

### ISSUE THREAD

**Issue by shilo-ey** [elementor/elementor#13434]
===========

# [Beta 4 UPDATED - PLEASE READ BELOW!](https://github.com/elementor/elementor/issues/13434#issuecomment-763790904)

===========
Hi Elementor Beta testers 👋, 

In a few weeks, we will be launching Elementor v3.1. This version will focus on some exciting new features and improvements aimed to streamline your marketing workflow, improve stability, accessibility, and enhance your performance. We would love for you to test Elementor v3.1 Beta and let us know what you think about the new features below:

- **A new landing page workflow**
- **Elementor Experiments** 
- **Compatibility Tag**
 

**Before you test, please make sure you are using:**
- Elementor Core v3.1 Beta
- Elementor Pro v3.0.9

## Landing Page Builder: A New Streamlined Way to Create and Manage Landing Pages

![Landing Pages (1)](https://user-images.githubusercontent.com/45009641/103291553-b310c280-49f4-11eb-915c-72f5c0e14c92.jpg)

Elementor is used by thousands of marketers and being able to quickly and efficiently create landing pages is crucial to their success. In this release, as well as future releases, we plan on adding many more features that will help save marketers time and improve the way they work.

### New landing page flow
Up until now, if you wanted to create a landing page, you had to select a page and choose the “Elementor Canvas” view in the page layout and filter through all of the page templates to find your desired ones. All that changes with the new landing page flow. 

To help streamline your workflow, we added the ability to create, manage, and organize all of your landing pages in one place. Now, under the “Templates” section, you’ll be able to save time by quickly accessing and creating landing pages as fast as two clicks.. 

![Landing Page](https://user-images.githubusercontent.com/45009641/103291669-f703c780-49f4-11eb-943a-fa39fbd9d902.gif)

### Dedicated landing page library
To give you more tools and help you create landing pages faster, we created a dedicated library of over 50 landing page templates. In the near future, we plan on adding many more new templates for you to choose from.

The introduction of this new landing page flow coupled with our landing page templates is part of our ongoing effort to provide our marketers with a faster way to build landing pages and scale their marketing activities.

Please note, Landing Pages is currently an experimental feature. You’ll need to activate it under the “Experiments” tab of your Elementor Settings. You can read more information about the Experiments down below

#### How does it work?

1. In your WordPress Dashboard go to Elementor > Settings > Experiments and turn on Landing Page experiment
2. In the sidebar find the “Templates” menu item, hover over it and click on the “Landing Page” button
3. Create a new Landing Page and make sure the Editor loads properly with a Canvas page layout applied
4. Right after the page has been created the Template Library will open and display the Landing Pages tab
5. Preview the templates you are offered, insert a template of your choice and make sure it is downloaded and applied correctly
6. Publish the Landing Page and make sure it displayed correctly in front-end
7. Make sure everything works as expected when trying to add several Landing Pages, changing their Page layout, deleting and restoring them back

 

## Elementor Experiments: Experience our Latest Cutting-edge Features Before they are Released

![Experiments](https://user-images.githubusercontent.com/45009641/103291584-c6bc2900-49f4-11eb-810c-bb4d70656d80.gif)

Since the release of Elementor 3.0, we have scaled our efforts of improving the way we test and release features. As part of that effort, we’re introducing a new Experiments capability that will allow you to experience, test, and provide feedback on some of the features before they are released. Not a big fan of the feature? Turn it off, and you’ll never have to see it again (as long as it’s an experiment). You’ll be able to try out these features in a safe environment and Influence future releases. 

Each experiment we introduce will contain a brief explanation about the feature, a status (beta/release candidate/dev) which indicates in what stage the feature is, and activity indication: green means active (even if the value Default is selected)
Grey means inactive (yes, even if the value Default is selected)

Please note that experiments may change or be removed based on the progress of the feature. We recommend that you use it with caution. Below are experiments you can expect to find in the upcoming release:

### Landing Pages
The new Landing Pages workflow is currently an experiment and will not be automatically turned on. Make sure it’s on “Active” if you would like to give it a try. 

### Accessibility Improvements
We’ve received some feedback that Elementor needs some A11y improvements, so we decided to focus on this area, starting with the Tabs widget which creates empty anchor tags, missing screen reader states per tab, and more. To fix this, we created an experiment designed to improve its accessibility. We removed empty `<a>` tags, added selected tab state, and improved keyboard navigation. You will be able to test the now accessible Tabs widget without damaging existing sites because it contains some markup changes.

#### How does it work?

- Create a new page using Elementor and drag in Tabs widget
- Play around with the widget to see if everything works as expected both on front-end and on Preview mode
- Use an Accessibility audit tool such as “[Wave](https://wave.webaim.org/)” or “[Google Lighthouse](https://developers.google.com/web/tools/lighthouse#devtools)” and run an audit on the frontend of this page
- Review the issues that were related to Tabs widget
- Go to Elementor > Settings > Experiments tab and turn on the Accessibility Improvements experiment to “Active” mode
- Go back to the page you created, view the page and run the audit again
- Compare the reports and validate that the issues were resolved
- Play around with the widget to see if everything works as expected both on Frontend and on Preview mode

### Improved Asset Loading
This experiment focuses on optimizing site speed and scoring. The [Improved Asset Loading experiment](https://developers.elementor.com/experiment-optimized-asset-loading/) will improve performance for the front-end, saving loading time per each page by loading parts of the infrastructure JS code only when needed on the page, which can result in up to 177 KB of JS files size reduction.

By introducing a new method [for attaching a JS handler to an element](https://developers.elementor.com/a-new-method-for-attaching-a-js-handler-to-an-element/) we are able to break the loading method into small chunks. users who use [`HTTP/2` protocol](https://www.cloudflare.com/website-optimization/http2/what-is-http2/) in their sites will benefit from efficient loading, due to Multiplexing which allows a browser to request all the assets in parallel. This results in a dramatic performance gain.

**Please note that in order for this experiment to work properly, you’ll have to use the Elementor Pro v3.0.9 version and above.** Currently, this experiment only applies to Elementor Core widget handlers (JS files), and the Swiper library in Elementor and Elementor Pro instances. Pro widgets handlers will be added in a later Pro version. 

#### How does it work?
1. In your existing site, upgrade to Elementor 3.1.0 Beta
2. Go to Elementor > Settings > Experiments tab and make sure the experiment indicator is inactive by default in existing website instances (grey indicator color)
3. Create a new page and insert a heading widget
4. View the network monitor when loading the page on the front-end and make sure that a file called: `preloaded-elements-handlers.js`, which contains all core widget handlers, is loaded on the page. Keep this tab open for later reference.
5. Create a new site and install Elementor 3.1.0 Beta
6. Go to Elementor > Settings > Experiments tab and make sure the experiment indicator is STILL inactive by default in new website instances (grey indicator color)
7. Create a new page and insert a heading widget
8. View the network monitor when loading the page on front-end and make sure that a file called `preloaded-elements-handlers.js` was not loaded on the page, while all of the widgets JS handles are loaded conditionally - when using each one of the following widgets on the page, a JS file with the same widget name should be loaded:
   - `accordion`
   - `alert`
   - `counter`
   - `progress`
   - `tabs`
   - `toggle`
   - `video`
   - `image-carousel`
   - `text-editor`
9. For example: When using the accordion widget, a file called: `accordion.[hash].bundle.min.js` should be loaded.
10. Add a widget that uses Swiper carousel (e.g Image Carousel widget) to the page and make sure it conditionally loads the Swiper library file called: `swiper.min.js`, based on the setting you set in this Experiment
11. Change the value of this experiment and make sure the behavior is consistent
12. Make sure everything works as expected on Frontend and on Preview mode

### DOM Improvements 
Elementor v3.0 [introduced a new improvement](https://developers.elementor.com/dom-improvements-ahead-html-wrappers-removal-from-v3-0/) - the removal and reduction of existing DOM elements. This feature allowed sites who used it to enjoy fewer elements per page, which resulted in better performance scoring and faster load times.
Due to the fact that this feature caused many sites that used custom code or 3rd party plugins to break, we decided to add it as an optional feature.

In this version, we migrated it to the Experiments tab and added a condition that states that this feature will be automatically activated in new sites. Don't worry, this feature currently won’t be active by default in existing sites. 

#### How does it work?
1. In your existing site, upgrade to Elementor 3.1.0 Beta
2. Go to Elementor > Settings > Experiments tab and make sure the experiment indicator is inactive by default in existing website instances (grey indicator color)
3. Create a new page and insert a template from the Template Library
4. Inspect the source code of the front-end to make sure `.elementor-inner`, `.elementor-row` and `.elementor-column-wrap` elements are present in your DOM.
5. Create a new site and install Elementor 3.1.0 Beta
6. Go to Elementor > Settings > Experiments tab and make sure the experiment indicator is active by default in new website instances (green indicator color)
7. Create a new page and insert a template from the Template Library
8. Inspect the source code of the front-end to make sure `.elementor-inner`, `.elementor-row` and `.elementor-column-wrap` elements are present in your DOM.
9. Change the value of this experiment and make sure the behavior is consistent
10. Make sure everything works as expected on Frontend and on Preview mode

## Compatibility Tag: See What’s Compatible with the Next Version of Elementor

![Compatibility Tag v1 (2)](https://user-images.githubusercontent.com/45009641/103292970-9c1f9f80-49f7-11eb-8833-bdcb36e8c8ea.jpg)

As part of its being an open-source project, Elementor includes an extensive developers API, enabling the creation of a vast eco-system of 3rd party extensions that build on and enhance Elementor’s capabilities. In fact, there are over 500 addons that extend Elementor’s capabilities. 

When 3rd party developers don’t update their extensions to make them compatible with the latest versions of Elementor, Elementor users can experience problems with their sites. 

To protect your sites from possible future conflicts, we created a Compatibility Tag that notifies users about plugins that are not compatible with their currently-installed version of Elementor. This will make it easier to debug issues if any arise.

**Calling all Elementor Plugin Developers!**
From v3.1.0 onwards, Elementor will show compatibility notice before upgrading. 
Please make sure that your plugin is compatible and up to date with the latest version of Elementor, based on the following information: https://developers.elementor.com/compatibility-tag/ 

**Please note:** any plugin with "Elementor" in its name will be automatically added to the list. For the benefit of our users we urge you to register your compatibility status.   

------

## More Improvements

* New: Elementor Experiments - Experience new features before they're officially released ([Developer Documentation](https://developers.elementor.com/elementor-experiments/))
* New: Compatibility Tag - Make sure your website plugins are compatible with Elementor ([Developer Documentation](https://developers.elementor.com/compatibility-tag/))
* Experiment: Landing Pages - Create beautiful landing pages in a streamlined workflow
* Experiment: Accessibility Improvements - Make Elementor widgets more accessible (may include markup changes) ([#13191](https://github.com/elementor/elementor/issues/13191))
* Experiment: Improved performance by loading JS and Swiper assets conditionally in frontend ([#8572](https://github.com/elementor/elementor/issues/8572), [Developer Documentation](https://developers.elementor.com/experiment-optimized-asset-loading))
* Tweak: Improved Tabs widget accessibility ([#11779](https://github.com/elementor/elementor/issues/11779), [#11561](https://github.com/elementor/elementor/issues/11561))
* Tweak: Added alignment options for Tabs widget ([#11997](https://github.com/elementor/elementor/issues/11997))
* Tweak: Updated Font Awesome icons library to v5.15.1 ([#12057](https://github.com/elementor/elementor/issues/12057))
* Tweak: Added "Custom" Columns Gap option in Section element ([#11978](https://github.com/elementor/elementor/issues/11978))
* Tweak: Added Border Radius support in Google Maps widget ([#11359](https://github.com/elementor/elementor/issues/11359))
* Tweak: Added dynamic capabilities to Tab Title control in Tabs widget ([#9710](https://github.com/elementor/elementor/issues/9710))
* Tweak: Added dynamic capabilities to Toggle Content control in Toggle widget ([#12405](https://github.com/elementor/elementor/issues/12405))
* Tweak: Added `em` unit to Border Radius control in Button widget ([#11561](https://github.com/elementor/elementor/issues/11561))
* Tweak: Introduced a new method for attaching a JS handler to an element ([Developer Documentation](https://developers.elementor.com/new-method-attaching-a-js-handler))
* Tweak: Refactored YouTube source to use YouTube API in Video widget
* Tweak: Added dynamic capabilities to Poster control in Video widget
* Tweak: Global dropdown controls are now displayed for all user roles
* Tweak: Improved browsers detection utility functionality
* Tweak: Added "Find an Expert" link to the Admin Dashboard Overview widget
* Tweak: Added the new Theme Builder as a Finder item
* Tweak: Added `+` icon in multi-select control for better UX
* Tweak: Improved Select2 controls load
* Tweak: Added Elementor Beta (Developer Edition) promotion in WordPress dashboard
* Tweak: Migrated DOM Improvements to Elementor Experiments
* Tweak: Removed redundant extra padding in responsive controls
* Tweak: Modified the Admin Bar menu item for Theme Builder app
* Fix: Some keyboards layout cannot open the keyboard shortcuts dialog ([#6145](https://github.com/elementor/elementor/issues/6145))
* Fix: Steps Divider is not vertically aligned in Multi Step Form widget  ([#12569](https://github.com/elementor/elementor/issues/12569))
* Fix: Double click needed to play YouTube video when Poster image exists in Video widget
* Fix: Autoplay option doesn't work in mobile devices when YouTube source is muted in Video widget
* Fix: YouTube End Time option doesn't work when Loop option is active in Video widget
* Fix: Incompatible variable names in Base Swiper class and Elementor Pro carousels
* Deprecated: See all deprecations to this version in our [Developers Deprecations Post](https://developers.elementor.com/v3-1-planned-deprecations/)

-----

## How to Install the Beta Version
If you still haven't enabled beta testing in Elementor, please follow these steps:

- Go to Elementor > Tools > Versions.
- Enable the beta tester feature.
- Go to the Dashboard > Updates and update to the latest beta version.
- You can also download the new version here: [downloads.wordpress.org/plugin/elementor.3.1.0-beta4.zip](https://downloads.wordpress.org/plugin/elementor.3.1.0-beta4.zip)

-----

## Notes
For bugs and remarks, please reply with a comment on this issue here on GitHub.

**Please keep this related to the current beta only! For any other issue or feature request, feel free to open a new issue.**

Designated time for release: `Janurary 26, 2021 (Estimated)`

Many thanks for your support and help!

Cheers,
_Elementor Team_

**Comment by fabianenzensperger** [comment]
Hmm. Ok cool, but Landing Pages as a main feature?

I do not know …

We already have a lot of patience with the Custom Brakepoints, and I understand that it takes longer, but why not process one of the other long-awaited open inquiries? (https://github.com/elementor/elementor/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc) PLEASE ELEMENTOR, listen to your community!

Having landing pages as a section is nice, but a key feature? The other things are very promising, however.

Before I rant for a long time and in more detail, I'll have a look at the new beta tonight.

**Comment by shilo-ey** [comment]
> Hmm. Ok cool, but Landing Pages as a main feature?
> 
> I do not know …
> 
> We already have a lot of patience with the Custom Brakepoints, and I understand that it takes longer, but why not process one of the other long-awaited open inquiries? ([elementor/elementor/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc](https://github.com/elementor/elementor/issues?q=is%3Aissue+is%3Aopen+sort%3Acomments-desc)) PLEASE ELEMENTOR, listen to your community!
> 
> Having landing pages as a section is nice, but a key feature? The other things are very promising, however.
> 
> Before I rant for a long time and in more detail, I'll have a look at the new beta tonight.

Hey @fabianenzensperger 

Thanks for your feedback.

The beta issue is supposed to tell about the new features in this version.
Actually, the link you attached is something that we are constantly following, and I can assure you that we will handle a lot of this issue in the upcoming releases.

One mistake we had in v3.0 is that we bundled a lot of code and infrastructure changes in one single (complex) release which caused conflicts and a very long debugging process. we wish to prevent this situation in the future, therefore our releases will be more compact and often, and will add features that are risky in a more gradual manner, by adding experiments and testing them before they are publicly available.



I suggest you to take a look at the experiment that handles #8572, which is an issue in the link you attached.

Thanks again and hope this release will meet your expectations.

**Comment by rodolphebertozzo** [comment]
I saw that v3.1 is scheduled for mid-February, which still leaves time to have a more than stable and performing version. Can you tell us how long before E. Pro will be available in beta?

**Comment by sermalefico** [comment]
Hi!

i use this code inside a shortcode to create a swiper slide for a post query before the new Improved Asset Loading experiment  and all is fine. Now with this new improvements swiper scripts now is no loading if no have any swiper widget, ok?

`
			
 var mySwiper<?php echo get_the_ID(); ?> = new Swiper ('.slide-archivo-<?php echo get_the_ID(); ?>', { 
      spaceBetween: 0,
	preloadImages: false,
        lazyLoading: false,
        observer: true,

    loop: true,
      pagination: {
        el: '.swiper-pagination-<?php echo get_the_ID(); ?>',
        clickable: true,
      },
      navigation: {
        nextEl: '.swiper-button-next-<?php echo get_the_ID(); ?>',
        prevEl: '.swiper-button-prev-<?php echo get_the_ID(); ?>',
      },
	  onSlideChangeEnd: function(swiper) {
            activeSlide = swiper.slides.eq( swiper.activeIndex );
            slideLazy = activeSlide.find( $( '.image-inview' ) );

            // disable Autoplay during Image preload
            if (slideLazy.hasClass('lazyload')) {
                swiper.stopAutoplay();
            }
        }
    });

`


Now i put this before the script [https://developers.elementor.com/experiment-optimized-asset-loading/](url)

`
const asyncSwiper = elementorFrontend.utils.swiper;
 
new asyncSwiper( swiperElement, swiperConfig ).then( ( newSwiperInstance ) => {
  console.log( 'New Swiper instance is ready: ', newSwiperInstance );
 
  mySwiper<?php echo get_the_ID(); ?> = newSwiperInstance;
} );

`

but my code doesn't works, swiper scripts isn't load. How is wrong?

this is the error:

`
jquery.min.js:2 Uncaught TypeError: Cannot read property 'swiper' of undefined
    at HTMLDocument.<anonymous> (portfolio:976)
    at e (jquery.min.js:2)
    at t (jquery.min.js:2)

`

Thans for this new improvement! if i get works this... will be awesome

**Comment by riadcitaku** [comment]
``` 
Undefined index: wp-content/plugins/elementor-pro/modules/popup/tag.php:87
Undefined index: popup wp-content/plugins/elementor-pro/modules/popup/tag.php:100
```

**Comment by FrostStudios** [comment]
All nice - yet your code is still way too bloated. 5 DIVs and 1 H1 Tag for a heading? Have a look at Oxygen ( I know you don't wanna hear that, but you will have to address that one day - the earlier the better )

But cudos for the conditional stuff.

**Comment by benstone326** [comment]
I think the Landing Page feature is totally unnecessary. I can't see what's the difference from the current page templates...
The new DOM feature seems really useful.

So please don't implement Landing Page library...

**Comment by freelancesam** [comment]
And yet.. still no sign of a decent form builder.

**Comment by DJviolin** [comment]
[This](https://elementor.com/help/how-will-elementor-work-with-wordpress-5-0-gutenberg-editor/) blog post stating:

> Will I be able to use Elementor inside Gutenberg or **vice versa**?
> Yes. Check out our Elementor Blocks for Gutenberg plugin in the WordPress repository.

Does this mean my PR request resolved: https://github.com/elementor/elementor/issues/11440

**Comment by ui-lawson** [comment]
> And yet.. still no sign of a decent form builder.

GravityForms ftw

**Comment by Jonathlee** [comment]
As most of the users has been hungry for performance and custom breakpoint improvement, I truly hope the Elementor team could really stop adding new features until major issue has been fixed such as;

![image](https://user-images.githubusercontent.com/696956/103368258-13415a80-4b02-11eb-8884-0ed1f171da90.png)
Growing file size of frontend js and css is getting scarier

1. Overbloated CSS files (the more feature you add, the more unused style added which increase file size, and more work to cater to custom breakpoints feature)
2. `.elementor-106 .elementor-element.elementor-element-194e2f5.elementor-wc-products ul.products li.product .button` is overkill and generated huge amount of unnecessary plus unused CSS. And most of all, those who wanted customize that particular element need to duplicate the same length of CSS to override this will take huge amount of effort to change this since you make this absolute including "div/ul".
3. As @FrostStudios mentioned, 5 divs (DIVception) is overkill. Any potential of mode such as"Highly Optimized Mode (Strictly for new installation only".
4. Limiting the maximum creation of columns to 10 is tough for those who wanted to modify columns via FLEXBOX (Flexbox Wrap) to add more than 10 columns.
5. Landing page workflow is nice, make it 3.2 is still better.
6. Modular CSS/JS loading is great.
7. Customize blog post loop would be amazing too 
8. Praying hard for Elementor Team to surprise us instead of just custom breakpoint, they are going to give us CSS Grid as well :D

CSS naming optimization

> As part of our effort to simplify and minimize the markup created by Elementor, we have decided to truncate the prefix we use for any Elementor CSS classes, from .elementor- to just .e-.
source: https://developers.elementor.com/updated-convention-creating-new-widgets-for-improved-performance/

By the way, may I know if there's any developer, designer who would like to override some styling which unfortunately fixated absolute selector in frontend.min.css as item 2 above?

I do aware that adding new features would be important, we hope the Elementor team would really focus to solve performance issue and make the website created with Elementor is superbly high-performance, accessibility friendly, customizable.

**Comment by benstone326** [comment]
> > And yet.. still no sign of a decent form builder.
> 
> 
> 
> GravityForms ftw

Yeah, but why would you install another plugin if you COULD do this in Elementor?

**Comment by mcrowell** [comment]
I'm seeing multiple 404 errors in the browser Console.
Below is a screenshot of the SourceMap 404 errors. 

![Elementor-404 SourceMaps](https://user-images.githubusercontent.com/7768009/103380042-a902c880-4ab5-11eb-91b5-a1f4b434cdf3.jpg)

**Comment by Ostii** [comment]
Typical software developers! Focusing on a new feature rather than all the issues with existing features.
Why, why, why?

**Comment by portrayaloflife** [comment]
I think this is great, love the experimental dock to safely roll out and test new features. Great stuff.

I agree though with others more customizable post loops and custom breaking points would be game-changers for site builders and are really what the community has been screaming for. Looking forward to having those options built-in! 

Or even make an Elemetnor add on for custom post loops and creating custom post types like pods or jet engine. Would be $$$ in your pocket. 

Keep up the great work!

**Comment by jonathansoltero** [comment]
I saw an Elementor Pro Beta Release email & got excited, I thought it was conditional logic for forms.

**Comment by rodolphebertozzo** [comment]
@jonathansoltero an email, can you show it?

**Comment by fabianenzensperger** [comment]
> @jonathansoltero an email, can you show it?


E-Mail from Elementor 30.12.2020 "Introducing Elementor v3.1 Beta"

"Dear Elementor beta testers,In just a few weeks, we will be launching Elementor v3.1, and today you get to try out its Beta version.

Here’s what you’ll find inside Elementor 3.1 Beta, based on your feedback:

- **Landing Pages** - New streamlined creation workflow and templates
- **Experiments** - Experience and control new features before they are officially released
- **Performance** - Conditional load of required assets
- **Accessibility Improvements** - Try the new accessible tabs widget

We can’t wait for you to test out these new features and let us know what you think.
 
Learn More Button
 
**Important Note!**
Found a bug? Please leave us a comment in the beta thread on GitHub.

Please keep the beta thread related to the current beta only. For other issues or feature requests - please open a new issue on Github.
 
Thank you for your support and help 🙏
The Elementor Team"

**Comment by portseif** [comment]
> As most of the users has been hungry for performance and custom breakpoint improvement, I truly hope the Elementor team could really stop adding new features until major issue has been fixed such as;
> 
> ![image](https://user-images.githubusercontent.com/696956/103368258-13415a80-4b02-11eb-8884-0ed1f171da90.png)
> Growing file size of frontend js and css is getting scarier
> 
> 1. Overbloated CSS files (the more feature you add, the more unused style added which increase file size, and more work to cater to custom breakpoints feature)
> 2. `.elementor-106 .elementor-element.elementor-element-194e2f5.elementor-wc-products ul.products li.product .button` is overkill and generated huge amount of unnecessary plus unused CSS. And most of all, those who wanted customize that particular element need to duplicate the same length of CSS to override this will take huge amount of effort to change this since you make this absolute including "div/ul".
> 3. As @FrostStudios mentioned, 5 divs (DIVception) is overkill. Any potential of mode such as"Highly Optimized Mode (Strictly for new installation only".
> 4. Limiting the maximum creation of columns to 10 is tough for those who wanted to modify columns via FLEXBOX (Flexbox Wrap) to add more than 10 columns.
> 5. Landing page workflow is nice, make it 3.2 is still better.
> 6. Modular CSS/JS loading is great.
> 7. Customize blog post loop would be amazing too
> 8. Praying hard for Elementor Team to surprise us instead of just custom breakpoint, they are going to give us CSS Grid as well :D
> 
> CSS naming optimization
> 
> > As part of our effort to simplify and minimize the markup created by Elementor, we have decided to truncate the prefix we use for any Elementor CSS classes, from .elementor- to just .e-.
> > source: https://developers.elementor.com/updated-convention-creating-new-widgets-for-improved-performance/
> 
> By the way, may I know if there's any developer, designer who would like to override some styling which unfortunately fixated absolute selector in frontend.min.css as item 2 above?
> 
> I do aware that adding new features would be important, we hope the Elementor team would really focus to solve performance issue and make the website created with Elementor is superbly high-performance, accessibility friendly, customizable.

This is the absolute truth. When I read “Developer Edition,” I got incredibly excited. However, landing pages should not be a developer’s feature. More granular control of loading assets, DOM improvements, and breakpoints are needed. 

Oxygen works like this:
1) you add a section or div
2) you add a div or column widget (2 or more divs) -OR- a widget (e.g., heading, text, etc.)
3) you can then nest other divs or widgets

*each widget and div then has flexbox, typology, spacing/size, and so much more. 

These should be the DOM elements presented. Complete granular control.

Also, a huge pet-peeve of mine is the ability to customize your CSS variables. It is such a pain for me to use your given --e- variables because most of the time, it is some random alphanumeric string.

If a user wants to use the swiper, tab, or accordion widgets, it loads in a div with a repeater widget that allows for easy customization/control. This is ideal.

I’m glad that Elementor is shedding light on the needed assets and DOM improvements, but please don’t start introducing a “Developer Edition” with Landing Page options. This is only taking what you currently have, changing it a little, and giving it a new name.

**Comment by shilo-ey** [comment]
Hi all,

First of all, thank you for your feedback.

@rodolphebertozzo  
> I saw that v3.1 is scheduled for mid-February, which still leaves time to have a more than stable and performing version. Can you tell us how long before E. Pro will be available in beta?
> 
> Another remark: the Marketing pages also match in the Wordpress pages section. Aren't they supposed to be separated?

There was a mistake with the release date and it was fixed, thanks.

The Pro version will be introduced soon as a separate beta version.

The Landing Pages currently are Pages, we are checking if keeping them that way is how we want to release this feature or changing it a bit. each option has its pros and cons.

-
> There's an issue appearing from Popup.

@sermalefico :  First, you need to make sure that `elementorFrontend` exists.
This object is being called from the `frontend.js` file, which is loaded from the footer.
Since you are trying to use a shortcode, probably you inserted it in the middle of the page (where the `frontend.js` file was not loaded yet). So, you get the error of `swiper is undefined`.
In addition, in case you are not using a widget handler, you need to listen to the following hook, which initializes `elementorFrontend.utils.swiper`
```
elementorFrontend.hooks.addAction( ‘components:init’, () => {
   // Do something that is based on the elementorFrontend object.
} );
```
-
@DJviolin 

-
@riadcitaku : Can you please share more light and steps to reproduce the errors?
-

@Jonathlee :
As for the file sizes, the file version should be 3.1.0-beta1 and not v3.0.15 we see from your screenshot. Only then, you will see a decrease in the file sizes.
Please make sure you are using the right version and you cleared your cache (server and browser side).
- CSS files - we are currently investigating how to approach the change without harming existing websites or various addons installed
- DOM - we are activating the Optimized DOM output by default to only new sites to prevent potential break or conflicts in existing sites. 
We will proceed these constant improvements in the next releases.

-
@mcrowell : Thanks! we're investigating this issue.
-

@portseif 
The Developer Edition is a different release channel that allow us to provide frequent releases with our development progress to extend the period that 3rd party developers have to adjust to changes.
The features that are introduced in the Developer Edition versions will not necessarily will be developer oriented.

As for bloat issues and performance upgrades - as mentioned before, we are working on these aspects and will continue to provide you with these kind of improvements from release to release.



**Please keep this related to the Current beta Only! any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.**

Thanks again to everyone.

**Comment by sermalefico** [comment]
@shilo-ey I don't understand how doing it works ... :(

**Comment by nerds-farm** [comment]
Select2 ajax search is not working on display selected values.
How to reproduce it:
- add widget Posts
- from Query Section search and select a term filter
- change section and return on Query Section

NO errors on console and no errors on Select2 ajax call (as display on screen).

![Screenshot_2021-01-02 Elementor Search Filter](https://user-images.githubusercontent.com/75386475/103456243-890d1800-4cf4-11eb-9565-9eb527c3c399.png)
![Schermata del 2021-01-02 12-10-33](https://user-images.githubusercontent.com/75386475/103456239-86122780-4cf4-11eb-95ca-6831b92a5202.png)
![Schermata del 2021-01-02 12-10-41](https://user-images.githubusercontent.com/75386475/103456241-87dbeb00-4cf4-11eb-9ee8-8d0b040b4abe.png)

**Comment by DorShahar** [comment]
Hi @mcrowell 
Thanks for your report. 

We had a bug on our end due to the Webpack upgrade, and we are fixing it now.

**Comment by shilo-ey** [comment]
> Select2 ajax search is not working on display selected values.
> How to reproduce it:
> 
> * add widget Posts
> * from Query Section search and select a term filter
> * change section and return on Query Section
> 
> NO errors on console and no errors on Select2 ajax call (as display on screen).
> 
> ![Screenshot_2021-01-02 Elementor Search Filter](https://user-images.githubusercontent.com/75386475/103456243-890d1800-4cf4-11eb-9565-9eb527c3c399.png)
> ![Schermata del 2021-01-02 12-10-33](https://user-images.githubusercontent.com/75386475/103456239-86122780-4cf4-11eb-95ca-6831b92a5202.png)
> ![Schermata del 2021-01-02 12-10-41](https://user-images.githubusercontent.com/75386475/103456241-87dbeb00-4cf4-11eb-9ee8-8d0b040b4abe.png)

@nerds-farm  Thanks!

It seem to work fine on our end. 
Can you please make sure you're using only Elementor and Elementor Pro to isolate potential conflicts?

Thanks

**Comment by rodolphebertozzo** [comment]
@shilo-ey, why did you move the HTML semantics optimization feature to the Experiences tab, since many sites have it enabled and it can be disabled anyway? Do you plan to make further improvements to avoid case on older sites using custom CSS to avoid case?

**Comment by shilo-ey** [comment]
> @shilo-ey I don't understand how doing it works ... :(

@sermalefico we updated the snippet:
```JS
elementorFrontend.on( 'components:init', () => {
   // Do something that is based on the elementorFrontend object.
} );
```

Check and let me know if it works for you

Thanks

**Comment by shilo-ey** [comment]
> @shilo-ey, why did you move the HTML semantics optimization feature to the Experiences tab, since many sites have it enabled and it can be disabled anyway? Do you plan to make further improvements to avoid case on older sites using custom CSS to avoid case?

@rodolphebertozzo we migrated this option to the Experiments tabs, so it will act the same, but will allow Elementor to control whether to activate it to existing or new sites by default.

Thanks

**Comment by pratikshasaman** [comment]
Hello,
I am not able to find the setting icon in the dynamic field lightbox with beta v3.1
Elementor Pro v3.0.9, the issue is resolved by switching to the Elementor current version.
No error in the console. 

Looking forward to hearing from you.

Thanks!

**Comment by Camilo517** [comment]
Popups do not open, there is no error in console

**Comment by shilo-ey** [comment]
> Hello,
> I am not able to find the setting icon in the dynamic field lightbox with beta v3.1
> Elementor Pro v3.0.9, the issue is resolved by switching to the Elementor current version.
> No error in the console. [a.cl.ly/KoulKGY7](https://a.cl.ly/KoulKGY7)
> 
> Looking forward to hearing from you.
> 
> Thanks!

@camilio517 @pratikshasaman

We are fixing this issue and it will be resolved in the next beta that is supposed to be released in the next few days.

Thanks

**Comment by Aleksandr0114** [comment]
Hi, our custom widgets uses  jquery-slick script. We've registered a script with the name: `jquery-slick`. Now, on the pages we've errors. Please, remove this notice: https://prnt.sc/whldql

**Comment by shilo-ey** [comment]
We just released a new beta version, please update your version to v3.1.0 Beta 2.

This beta release includes:

- Tweak: Add vertical alignment for Tabs widget (#11997)
- Tweak: Converted Elementor Landing Pages to Custom Post type
- Tweak: Added an option to choose Text element HTML tag in Divider widget
- Tweak: Added more Webpack optimization to improve performance
- Tweak: Improved browser detection mechanism
- Tweak: Updated Experiment status for Landing Pages and Accessibility Improvements
- Fix: Duplicate items in responsive view of Tabs widget
- Fix: Unable to open Dynamic tag controls popover
- Fix: Columns Gap control is not working when Optimized DOM output mode is inactive
- Fix: JS source map files caused console warnings
- Fix: Some dynamic controls were not available
- Fix: Console error was thrown when activating Motion Effects with Legacy DOM mode


**Please keep this related to the Current beta Only!** any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

**Thanks again to everyone.**

**Comment by rodolphebertozzo** [comment]
https://github.com/elementor/elementor/issues/11499

**Comment by portrayaloflife** [comment]
Is the social media icons bug going to be fixed?

**Comment by fabianenzensperger** [comment]
Good update.

@shilo-ey I have a question: What exactly do I do with the landing pages? I have created one and how do I access it on the website? When I go to "View" I get: "The content was not found". What's the purpose and what am I doing wrong?

Is this like a blog or a single site with Gutenberg Content?

**Comment by nerds-farm** [comment]
> > Select2 ajax search is not working on display selected values.
> > How to reproduce it:
> > 
> > * add widget Posts
> > * from Query Section search and select a term filter
> > * change section and return on Query Section
> > 
> > NO errors on console and no errors on Select2 ajax call (as display on screen).
> > ![Screenshot_2021-01-02 Elementor Search Filter](https://user-images.githubusercontent.com/75386475/103456243-890d1800-4cf4-11eb-9565-9eb527c3c399.png)
> > ![Schermata del 2021-01-02 12-10-33](https://user-images.githubusercontent.com/75386475/103456239-86122780-4cf4-11eb-95ca-6831b92a5202.png)
> > ![Schermata del 2021-01-02 12-10-41](https://user-images.githubusercontent.com/75386475/103456241-87dbeb00-4cf4-11eb-9ee8-8d0b040b4abe.png)
> 
> @nerds-farm Thanks!
> 
> It seem to work fine on our end.
> Can you please make sure you're using only Elementor and Elementor Pro to isolate potential conflicts?
> 
> Thanks

I tested also with simple Posts (post_type "post") and related Categories.
I create a new fresh page with only Portfolio widget.
The bug appear with 3.1.0-beta2 and also 3.2.0-dev1.

I activate only Elementor plugins, no extra addons.
Here my system info.
[system-info-localhost-08-01-2021.txt](https://github.com/elementor/elementor/files/5786743/system-info-localhost-08-01-2021.txt)

**Comment by kochhase** [comment]
why are there  3.1.0-beta2 and also 3.2.0-dev1. ? is there an difference if i m using the Elementor Beta plugin or not?

**Comment by Jonathlee** [comment]
After these update it brick most of the pages including the store page : https://store.jonathlee.com/shop/
most of the pages show error 404 not found.

**Comment by Camilo517** [comment]
The widget "product images" does not work when you use an image in webp format

**Comment by shilo-ey** [comment]
We just released a new beta version, please update your version to v3.1.0 Beta 3.

This beta release includes:

- Tweak: Added the option to add custom menu items in Site Settings ([Developer Documentation](https://github.com/elementor/elementor/pull/13243))
- Tweak: Added compatibility status to plugin System Info
- Tweak: Added “Experiments” item to the Finder
- Tweak: Added tooltips to the Experiment indicators in Experiments screen
- Tweak: Removed “Pages” tab from the Template Library modal when editing Landing Pages
- Tweak: Added a Category filter to the Landing Pages tab in the Template Library modal
- Tweak: Activate Landing Pages experiment by default
- Fix: Change Page Layout doesn't update the preview on the first attempt (#13245)
- Fix: Landing Pages menu item directs to the wrong page
- Fix: Wrong value in Landing Pages dashboard Type column
- Fix: Add New Landing Page Finder item shown twice
- Fix: Viewing a new Landing Page redirects to the wrong permalink
- Fix: Can’t rearrange items between sections in Navigator since WordPress 5.6 update 
- Fix: Compatibility Tag contains the plugin directory path and not plugin name
- Fix: Typo in Site Settings delete screen
- Fix: Redundant spacing is added to WYSIWYG control if rich editing is disabled 
- Fix: ‘+’ icon alignment glitch in Select2 control
- Fix: Editor Autoplay is not working consistently in Image Carousel widget
- Fix: Select2 Query control rendered multiple types caused the control to break 
- Fix: Console errors are thrown when entering the Revisions menu in edge cases
- Fix: Update Post Title via Site Settings removes style from Post Title widget

**Please keep this related to the Current beta Only!** any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

**Thanks again to everyone.**

**Comment by Jonathlee** [comment]
Need you guys please check with Woocommerce. 

I created Woocommerce store with Elementor beta and Hello Theme : https://store.jonathlee.com/shop/
In the store page, each product did not linked back to respective individual product URL.
Using the latest woocomerce 4.9.

**Comment by shilo-ey** [comment]
@Aleksandr0114
> Hi, our custom widgets uses jquery-slick script. We've registered a script with the name: `jquery-slick`. Now, on the pages we've errors. Please, remove this notice: [prnt.sc/whldql](https://prnt.sc/whldql)

Thanks. Slick library was deprecated several versions ago. you can change the slick to swiper and that way you will be able to remove this note.



@portrayaloflife
> Is the social media icons bug going to be fixed?

Can you please be more specific?


@fabianenzensperger
> Good update.
> 
> @shilo-ey I have a question: What exactly do I do with the landing pages? I have created one and how do I access it on the website? When I go to "View" I get: "The content was not found". What's the purpose and what am I doing wrong?
> 
> Is this like a blog or a single site with Gutenberg Content?

It was a bug that was fixed due to your report, thanks. 
It's just another template type with more streamlined workflow and a dedicated Template Library.





> > > Select2 ajax search is not working on display selected values.
> > > How to reproduce it:
> > > 
> > > * add widget Posts
> > > * from Query Section search and select a term filter
> > > * change section and return on Query Section
> > > 
> > > NO errors on console and no errors on Select2 ajax call (as display on screen).
> > > ![Screenshot_2021-01-02 Elementor Search Filter](https://user-images.githubusercontent.com/75386475/103456243-890d1800-4cf4-11eb-9565-9eb527c3c399.png)
> > > ![Schermata del 2021-01-02 12-10-33](https://user-images.githubusercontent.com/75386475/103456239-86122780-4cf4-11eb-95ca-6831b92a5202.png)
> > > ![Schermata del 2021-01-02 12-10-41](https://user-images.githubusercontent.com/75386475/103456241-87dbeb00-4cf4-11eb-9ee8-8d0b040b4abe.png)
> > 
> > 
> > @nerds-farm Thanks!
> > It seem to work fine on our end.
> > Can you please make sure you're using only Elementor and Elementor Pro to isolate potential conflicts?
> > Thanks
> 
> I tested also with simple Posts (post_type "post") and related Categories.
> I create a new fresh page with only Portfolio widget.
> The bug appear with 3.1.0-beta2 and also 3.2.0-dev1.
> 
> I activate only Elementor plugins, no extra addons.
> Here my system info.
> [system-info-localhost-08-01-2021.txt](https://github.com/elementor/elementor/files/5786743/system-info-localhost-08-01-2021.txt)

@nerds-farm - Thanks, please try v3.1.0-beta3 and let me know if it sorted the issue for you.



> why are there 3.1.0-beta2 and also 3.2.0-dev1. ? is there an difference if i m using the Elementor Beta plugin or not?

Beta - this is 3.1.0 that is going to be released.
Dev - updating channel of v3.1.0 and more features. more features will be added soon.




> Need you guys please check with Woocommerce.
> 
> I created Woocommerce store with Elementor beta and Hello Theme : [store.jonathlee.com/shop](https://store.jonathlee.com/shop/)
> In the store page, each product did not linked back to respective individual product URL.
> Using the latest woocomerce 4.9.
> 
> I tried rollback to previous stable version and the Product page work.

Please try using Elementor 3.1.0-beta3 and the latest stable Pro version (v3.0.9)
It seems like it's working on our end. 
If the issue persists in the Dev-edition version - please create a dedicated issue.


Thank you all!

**Comment by Jonathlee** [comment]
![image](https://user-images.githubusercontent.com/696956/104557777-c40c3580-567c-11eb-80b3-d323665df0d8.png)
I received this notification on `There is a new version of Elementor available. View version 3.2.0-dev1 details or update now. `
May I know if this is another beta?

**Comment by shilo-ey** [comment]
'Version 3.2.0-dev1' is the dev version.

Use Elementor > Tools > Version Control and change to `v3.1.0-beta3` (and make sure you are using latest stable Pro version)

thanks

**Comment by Aleksandr0114** [comment]
@shilo-ey ,
We've many themes with our custom widgets. We cannot switch from slick to swiper in old themes. We've a special CSS for each theme for slick. This notice will be visible to everyone  who uses our theme.

**Comment by fabianenzensperger** [comment]
I recorded a video with the problem I still have with the Landing Pages. I honestly don't understand them.

https://youtu.be/zyCgyquZwbA

**Comment by rodolphebertozzo** [comment]
Hy,
The version seems stable to me ;)
A small remark: in the beta 3 chengelog, the link next to "Tweak: Added the option to add custom menu items in Site Settings" refers to the PR and not to the normally edited article on the developer documentation.

**Comment by portrayaloflife** [comment]
@shilo-ey Social icons issue here >> [#13122](https://github.com/elementor/elementor/issues/13122)

**Comment by kangarko** [comment]
@shilo-ey on beta 2 and beta 3 when you have "Improved Asset Loading" enabled it breaks popups on Android (those popups that are toggled by clicking a button)

**Comment by shilo-ey** [comment]
@Aleksandr0114

> @shilo-ey ,
> We've many themes with our custom widgets. We cannot switch from slick to swiper in old themes. We've a special CSS for each theme for slick. This notice will be visible to everyone who uses our theme.

In each version, there are deprecations of functions, methods, hooks etc. 
You can get [updated on the list here](https://developers.elementor.com/?s=deprecations) from v2.6

We have 3 phases: 
- Soft deprecation
- Hard deprecation (throws a notice when used with WP_DEBUG mode on)
- Deletion - will throw errors if you update to later versions that use custom functionalities such as this one

See more information about this change here: https://developers.elementor.com/elementor-2-7-moving-sliders-from-slick-to-swiper/

 

@kangarko
> @shilo-ey on beta 2 and beta 3 when you have "Improved Asset Loading" enabled it breaks popups on Android (those popups that are toggled by clicking a button)

Can you please share more information on how to reproduce this issue?


@fabianenzensperger

> I recorded a video with the problem I still have with the Landing Pages. I honestly don't understand them.
> 
> [youtu.be/zyCgyquZwbA](https://youtu.be/zyCgyquZwbA)


Can you share your system info please?

Thanks!

**Comment by kangarko** [comment]
@shilo-ey to reproduce broken popup, simply create a popup then create a page with a button linking to open that popup and it won't work on mobile

**Comment by fabianenzensperger** [comment]
> @Aleksandr0114
> 
> > @shilo-ey ,
> > We've many themes with our custom widgets. We cannot switch from slick to swiper in old themes. We've a special CSS for each theme for slick. This notice will be visible to everyone who uses our theme.
> 
> In each version, there are deprecations of functions, methods, hooks etc.
> You can get [updated on the list here](https://developers.elementor.com/?s=deprecations) from v2.6
> 
> We have 3 phases:
> 
> * Soft deprecation
> * Hard deprecation (throws a notice when used with WP_DEBUG mode on)
> * Deletion - will throw errors if you update to later versions that use custom functionalities such as this one
> 
> See more information about this change here: https://developers.elementor.com/elementor-2-7-moving-sliders-from-slick-to-swiper/
> 
> @kangarko
> 
> > @shilo-ey on beta 2 and beta 3 when you have "Improved Asset Loading" enabled it breaks popups on Android (those popups that are toggled by clicking a button)
> 
> Can you please share more information on how to reproduce this issue?
> 
> @fabianenzensperger
> 
> > I recorded a video with the problem I still have with the Landing Pages. I honestly don't understand them.
> > [youtu.be/zyCgyquZwbA](https://youtu.be/zyCgyquZwbA)
> 
> Can you share your system info please?
> 
> Thanks!
					
== Server Environment ==
	Operating System: Linux
	Software: Apache
	MySQL version: MySQL Community Server (GPL) v5.7.32
	PHP Version: 7.4.14
	PHP Max Input Vars: 5000
	PHP Max Post Size: 64M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 5.6
	Site URL: https://wp56.fabianenzensperger.de
	Home URL: https://wp56.fabianenzensperger.de
	WP Multisite: No
	Max Upload Size: 64 MB
	Memory limit: -1
	Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
	Language: de-DE
	Timezone: Europe/Berlin
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor Child
	Version: 1.0.1
	Author: Elementor Team
	Child Theme: Yes
	Parent Theme Name: Hello Elementor
	Parent Theme Version: 2.3.1
	Parent Theme Author: Elementor Team

== User ==
	Role: administrator
	WP Profile lang: de_DE
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 11_1_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36

== Active Plugins ==
	Custom Post Type UI
		Version: 1.8.2
		Author: WebDevStudios

	Elementor
		Version: 3.1.0-beta3
		Author: Elementor.com

	Elementor Pro
		Version: 3.0.9
		Author: Elementor.com

	Envato Elements
		Version: 2.0.7
		Author: Envato


== Elementor - Compatibility Tag ==
	
	Elementor Pro: header_not_exists

== Elements Usage ==
	
	footer : 1
		button : 1
		form : 1
		heading : 7
		image : 1
		nav-menu : 1
		text-editor : 3
	header : 1
		button : 1
		image : 1
		nav-menu : 1
	section : 20
		animated-headline : 1
		button : 13
		divider : 9
		flip-box : 10
		heading : 44
		icon-list : 3
		image : 16
		image-carousel : 5
		slides : 2
		spacer : 9
		text-editor : 28
		video : 2
	wp-page : 6
		animated-headline : 2
		button : 15
		call-to-action : 4
		divider : 9
		flip-box : 10
		heading : 76
		icon-box : 3
		icon-list : 3
		image : 16
		image-carousel : 5
		slides : 2
		spacer : 12
		text-editor : 31
		video : 2
	landing-page : 0
		button : 8
		call-to-action : 8
		countdown : 1
		form : 6
		gallery : 1
		google_maps : 2
		heading : 101
		icon : 17
		icon-box : 25
		icon-list : 9
		image : 21
		image-box : 6
		image-carousel : 1
		menu-anchor : 1
		price-list : 7
		reviews : 1
		social-icons : 2
		spacer : 31
		star-rating : 3
		testimonial : 3
		text-editor : 32
		video : 3


== Log ==
	
Log: showing 7 of 72020-12-17 23:10:07 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.0.14',
  'to' => '3.1.0-dev2',
)]
2020-12-18 11:50:12 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.0-dev2',
  'to' => '3.1.0-dev3',
)]
2020-12-21 18:35:45 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.0-dev3',
  'to' => '3.1.0-dev4',
)]
2020-12-29 18:44:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.0-dev4',
  'to' => '3.1.0-beta1',
)]
2020-12-29 18:44:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.0.8',
  'to' => '3.0.9',
)]
2021-01-07 18:40:19 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.0-beta1',
  'to' => '3.1.0-beta2',
)]
2021-01-14 01:07:59 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.1.0-beta2',
  'to' => '3.1.0-beta3',
)]

JS: showing 3 of 3JS: 2020-12-17 22:10:09 [error X 20][https://wp56.fabianenzensperger.de/wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=5.6:2:31670] Cannot read property \'landingPagesHasPages\' of undefined 
JS: 2020-12-17 22:10:13 [error X 7][https://wp56.fabianenzensperger.de/wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=5.6:2:31670] elementorAdmin is not defined 
JS: 2020-12-17 22:35:01 [error X 6][https://wp56.fabianenzensperger.de/wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=5.6:2:31670] Cannot read property \'option_enabled\' of undefined 

PHP: showing 3 of 3PHP: 2020-12-17 23:28:01 [notice X 17][/homepages/42/d580247425/htdocs/sonstiges/56/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-12-23 16:59:07 [notice X 2][/homepages/42/d580247425/htdocs/sonstiges/56/wp-content/plugins/elementor/core/document-types/page.php::96] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2020-12-24 01:07:26 [notice X 1][/homepages/42/d580247425/htdocs/sonstiges/56/wp-content/plugins/elementor/core/document-types/page.php::96] Undefined index: settings [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

**Comment by shilo-ey** [comment]
@kangarko 
> @shilo-ey to reproduce broken popup, simply create a popup then create a page with a button linking to open that popup and it won't work on mobile

Thanks, unfortunately, we can't reproduce this case, both by clicking on a button via Dynamic Tag and via click on a selector.

Are you sure you added the display conditions to be in the page you are trying to trigger the Popup?




> > @Aleksandr0114
> > > @shilo-ey ,
> > > We've many themes with our custom widgets. We cannot switch from slick to swiper in old themes. We've a special CSS for each theme for slick. This notice will be visible to everyone who uses our theme.
> > 
> > 
> > In each version, there are deprecations of functions, methods, hooks etc.
> > You can get [updated on the list here](https://developers.elementor.com/?s=deprecations) from v2.6
> > We have 3 phases:
> > 
> > * Soft deprecation
> > * Hard deprecation (throws a notice when used with WP_DEBUG mode on)
> > * Deletion - will throw errors if you update to later versions that use custom functionalities such as this one
> > 
> > See more information about this change here: [developers.elementor.com/elementor-2-7-moving-sliders-from-slick-to-swiper](https://developers.elementor.com/elementor-2-7-moving-sliders-from-slick-to-swiper/)
> > @kangarko
> > > @shilo-ey on beta 2 and beta 3 when you have "Improved Asset Loading" enabled it breaks popups on Android (those popups that are toggled by clicking a button)
> > 
> > 
> > Can you please share more information on how to reproduce this issue?
> > @fabianenzensperger
> > > I recorded a video with the problem I still have with the Landing Pages. I honestly don't understand them.
> > > [youtu.be/zyCgyquZwbA](https://youtu.be/zyCgyquZwbA)
> > 
> > 
> > Can you share your system info please?
> > Thanks!
> 
> == Server Environment ==
> Operating System: Linux
> Software: Apache
> MySQL version: MySQL Community Server (GPL) v5.7.32
> PHP Version: 7.4.14
> PHP Max Input Vars: 5000
> PHP Max Post Size: 64M
> GD Installed: Yes
> ZIP Installed: Yes
> Write Permissions: All right
> Elementor Library: Connected
> 
> == WordPress Environment ==
> Version: 5.6
> Site URL: [wp56.fabianenzensperger.de](https://wp56.fabianenzensperger.de)
> Home URL: [wp56.fabianenzensperger.de](https://wp56.fabianenzensperger.de)
> WP Multisite: No
> Max Upload Size: 64 MB
> Memory limit: -1
> Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
> Language: de-DE
> Timezone: Europe/Berlin
> Debug Mode: Inactive
> 
> == Theme ==
> Name: Hello Elementor Child
> Version: 1.0.1
> Author: Elementor Team
> Child Theme: Yes
> Parent Theme Name: Hello Elementor
> Parent Theme Version: 2.3.1
> Parent Theme Author: Elementor Team
> 
> == User ==
> Role: administrator
> WP Profile lang: de_DE
> User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 11_1_0) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/87.0.4280.141 Safari/537.36
> 
> == Active Plugins ==
> Custom Post Type UI
> Version: 1.8.2
> Author: WebDevStudios
> 
> ```
> Elementor
> 	Version: 3.1.0-beta3
> 	Author: Elementor.com
> 
> Elementor Pro
> 	Version: 3.0.9
> 	Author: Elementor.com
> 
> Envato Elements
> 	Version: 2.0.7
> 	Author: Envato
> ```
> 
> == Elementor - Compatibility Tag ==
> 
> ```
> Elementor Pro: header_not_exists
> ```
> 
> == Elements Usage ==
> 
> ```
> footer : 1
> 	button : 1
> 	form : 1
> 	heading : 7
> 	image : 1
> 	nav-menu : 1
> 	text-editor : 3
> header : 1
> 	button : 1
> 	image : 1
> 	nav-menu : 1
> section : 20
> 	animated-headline : 1
> 	button : 13
> 	divider : 9
> 	flip-box : 10
> 	heading : 44
> 	icon-list : 3
> 	image : 16
> 	image-carousel : 5
> 	slides : 2
> 	spacer : 9
> 	text-editor : 28
> 	video : 2
> wp-page : 6
> 	animated-headline : 2
> 	button : 15
> 	call-to-action : 4
> 	divider : 9
> 	flip-box : 10
> 	heading : 76
> 	icon-box : 3
> 	icon-list : 3
> 	image : 16
> 	image-carousel : 5
> 	slides : 2
> 	spacer : 12
> 	text-editor : 31
> 	video : 2
> landing-page : 0
> 	button : 8
> 	call-to-action : 8
> 	countdown : 1
> 	form : 6
> 	gallery : 1
> 	google_maps : 2
> 	heading : 101
> 	icon : 17
> 	icon-box : 25
> 	icon-list : 9
> 	image : 21
> 	image-box : 6
> 	image-carousel : 1
> 	menu-anchor : 1
> 	price-list : 7
> 	reviews : 1
> 	social-icons : 2
> 	spacer : 31
> 	star-rating : 3
> 	testimonial : 3
> 	text-editor : 32
> 	video : 3
> ```
> 
> == Log ==
> 
> Log: showing 7 of 72020-12-17 23:10:07 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor',
> 'from' => '3.0.14',
> 'to' => '3.1.0-dev2',
> )]
> 2020-12-18 11:50:12 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor',
> 'from' => '3.1.0-dev2',
> 'to' => '3.1.0-dev3',
> )]
> 2020-12-21 18:35:45 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor',
> 'from' => '3.1.0-dev3',
> 'to' => '3.1.0-dev4',
> )]
> 2020-12-29 18:44:09 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor',
> 'from' => '3.1.0-dev4',
> 'to' => '3.1.0-beta1',
> )]
> 2020-12-29 18:44:09 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor Pro',
> 'from' => '3.0.8',
> 'to' => '3.0.9',
> )]
> 2021-01-07 18:40:19 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor',
> 'from' => '3.1.0-beta1',
> 'to' => '3.1.0-beta2',
> )]
> 2021-01-14 01:07:59 [info] Elementor data updater process has been completed. [array (
> 'plugin' => 'Elementor',
> 'from' => '3.1.0-beta2',
> 'to' => '3.1.0-beta3',
> )]
> 
> JS: showing 3 of 3JS: 2020-12-17 22:10:09 [error X 20][https://wp56.fabianenzensperger.de/wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=5.6:2:31670] Cannot read property 'landingPagesHasPages' of undefined
> JS: 2020-12-17 22:10:13 [error X 7][https://wp56.fabianenzensperger.de/wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=5.6:2:31670] elementorAdmin is not defined
> JS: 2020-12-17 22:35:01 [error X 6][https://wp56.fabianenzensperger.de/wp-admin/load-scripts.php?c=1&loadchunk_0=jquery-core,jquery-migrate,utils&ver=5.6:2:31670] Cannot read property 'option_enabled' of undefined
> 
> PHP: showing 3 of 3PHP: 2020-12-17 23:28:01 [notice X 17][/homepages/42/d580247425/htdocs/sonstiges/56/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined index: data [array (
> 'trace' => '
> #0: Elementor\Core\Logger\Manager -> shutdown()
> ',
> )]
> PHP: 2020-12-23 16:59:07 [notice X 2][/homepages/42/d580247425/htdocs/sonstiges/56/wp-content/plugins/elementor/core/document-types/page.php::96] Trying to access array offset on value of type null [array (
> 'trace' => '
> #0: Elementor\Core\Logger\Manager -> shutdown()
> ',
> )]
> PHP: 2020-12-24 01:07:26 [notice X 1][/homepages/42/d580247425/htdocs/sonstiges/56/wp-content/plugins/elementor/core/document-types/page.php::96] Undefined index: settings [array (
> 'trace' => '
> #0: Elementor\Core\Logger\Manager -> shutdown()
> ',
> )]

@fabianenzensperger please try changing the permalinks to Plain and let me know if it solves the issue for you.

Thanks

**Comment by fabianenzensperger** [comment]
> 
> @fabianenzensperger please try changing the permalinks to Plain and let me know if it solves the issue for you.
> 
> Thanks

That worked, thanks!

**Comment by shilo-ey** [comment]
@fabianenzensperger And if you revert the permalinks back the issue reproduces?

**Comment by fabianenzensperger** [comment]
> @fabianenzensperger And if you revert the permalinks back the issue reproduces?

Thanks for asking, now it doesn't matter which permalink structure I use. It works with everyone now. I don't know why it didn't work before.

**Comment by jazir5** [comment]
I thought this was releasing today, when is it releasing @shilo-ey ?

**Comment by fabianenzensperger** [comment]
> I thought this was releasing today, when is it releasing @shilo-ey ?

Hmm Why did you think that? Who ever said that?

**Comment by jazir5** [comment]
It says release date January 19th at the bottom of the beta release post...

**Comment by fabianenzensperger** [comment]
> It says release date January 19th at the bottom of the beta release post...

Oh ok yeah. But only: Probably! 👀

**Comment by jazir5** [comment]
Well that's why I asked the question in the first place

**Comment by shilo-ey** [comment]
Hi all!

We just released a new beta version, please update your version to v3.1.0 Beta 4.

Due to several bugs found with your help, we decided to postpone the release date to next week.
Please check the new version and make sure everything works flawlessly.

**This beta release includes:**

- Tweak: Updated `eicons` library to v5.10.0
- Fix: UI glitch in the Library modal footer
- Fix: 404 page displayed when trying to access Landing Pages
- Fix: Carousel functionality didn't work if Optimized DOM Mode was disabled
- Fix: Column height is not correct when Optimized DOM experiment is inactive in IE11
- Fix: Wrong translation function caused errors in the Revisions panel
- Fix: Widgets empty state is not visible in WordPress 5.6

**Please keep this related to the Current beta Only!** any other [Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

**Thanks again to everyone.**

**Comment by portrayaloflife** [comment]
@shilo-ey Is the social icons bug fixed? Don't see it in the changelog

**Comment by shilo-ey** [comment]
> @shilo-ey Is the social icons bug fixed? Don't see it in the changelog

Please keep this thread related to the Beta only

**Comment by portrayaloflife** [comment]
@shilo-ey It's a bug that's not fixed in the beta that has now been almost 3 months unfixed... Not sure how that isn't related.

**Comment by sermalefico** [comment]
> @shilo-ey It's a bug that's not fixed in the beta that has now been almost 3 months unfixed... Not sure how that isn't related.

Yes in elementor blogs pages have the same bug. https://elementor.com/blog/redirect-url-wordpress/ 

check share section on the bottom post

**Comment by shilo-ey** [comment]
> > @shilo-ey It's a bug that's not fixed in the beta that has now been almost 3 months unfixed... Not sure how that isn't related.
> 
> Yes in elementor blogs pages have the same bug. [elementor.com/blog/redirect-url-wordpress](https://elementor.com/blog/redirect-url-wordpress/)
> 
> check share section on the bottom post

@sermalefico 

Do you have a comparison before/after?

Or even a JSON template export for us to examine?

Is it browser related or happens on all devices?

**Comment by sermalefico** [comment]
Add a advise to 

[https://wordpress.org/support/topic/incompatibility-with-future-elementor-update-custom-codes/](url)

**Comment by sermalefico** [comment]
> > > @shilo-ey It's a bug that's not fixed in the beta that has now been almost 3 months unfixed... Not sure how that isn't related.
> > 
> > 
> > Yes in elementor blogs pages have the same bug. [elementor.com/blog/redirect-url-wordpress](https://elementor.com/blog/redirect-url-wordpress/)
> > check share section on the bottom post
> 
> @sermalefico
> 
> Do you have a comparison before/after?
> 
> Or even a JSON template export for us to examine?
> 
> Is it browser related or happens on all devices?

This occurs when using a CSS minifier, such as autoptimize or Fastvelocity minify. For some reason CSS variables have some space that in this specific widget is broken on the frontend. Yes, all devices.

**Comment by fabianenzensperger** [comment]
@shilo-ey Coming tomorrow? :)

**Comment by perks98** [comment]
Elementor 3.1 is now public it seems

**Comment by shilo-ey** [comment]
> > > > @shilo-ey It's a bug that's not fixed in the beta that has now been almost 3 months unfixed... Not sure how that isn't related.
> > > 
> > > 
> > > Yes in elementor blogs pages have the same bug. [elementor.com/blog/redirect-url-wordpress](https://elementor.com/blog/redirect-url-wordpress/)
> > > check share section on the bottom post
> > 
> > 
> > @sermalefico
> > Do you have a comparison before/after?
> > Or even a JSON template export for us to examine?
> > Is it browser related or happens on all devices?
> 
> This occurs when using a CSS minifier, such as autoptimize or Fastvelocity minify. For some reason CSS variables have some space that in this specific widget is broken on the frontend. Yes, all devices.

@sermalefico 
Thanks! will check again




> Add a advise to
> 
> [https://wordpress.org/support/topic/incompatibility-with-future-elementor-update-custom-codes/](url)

Cool, thanks for letting us know, we will check if the conflict is due to an implementation issue on our side.


@fabianenzensperger Yes, today the version is rolling out. 

Thanks for your feedback.

Will update here once it's officially released.

**Comment by shilo-ey** [comment]
Thank you all for your precious comments and reports!

The version is now publicly available via WordPress repository or via your plugins dashboard.

If you encounter any new issue regarding this version please create a new issue report.

Thanks again!
